src/Hub.cpp

```diff
@@ -289,7 +328,42 @@ const sc2::Point3D* Hub::GetNextExpansion() {
         return nullptr;
 
     it->owner = Owner::CONTESTED;
-    return &(it->town_hall_location);
+    return &(*it);
+}
+
+void Hub::UpdateExpansionOwner(const sc2::Unit& unit_) {
+    for (auto& i : m_expansions) {
+        if (std::floor(i.town_hall_location.x) != std::floor(unit_.pos.x) ||
+                std::floor(i.town_hall_location.y) != std::floor(unit_.pos.y))
+            continue;
+        
+        if (!unit_.is_alive) {
```

Comment:
Imho, this method is too generic. There should be separate clean logic regarding capturing and loosing regions.
The context gives enough info regarding which event actually happened so we shouldn't mix everything inside a new call.

Reply:
Separated methods. Looks much cleaner.